src/components/canvas-tools/geometry-tool.tsx

```diff
@@ -123,16 +130,21 @@ class GeometryToolComponentImpl extends BaseComponent<IProps, IState> {
   }
 
   private initializeContent() {
-    const { model: { content }, readOnly } = this.props;
+    const { model: { content } } = this.props;
     if ((content.type !== "Geometry") || !this.state.elementId) { return; }
 
-    const board = content.initializeBoard(this.state.elementId, this.handleCreateElement);
-    if (board) {
-      this.handleCreateBoard(board);
+    const domElt = document.getElementById(this.state.elementId);
+    const eltBounds = domElt && domElt.getBoundingClientRect();
+    // JSXGraph fails hard if the DOM element doesn't exist or has zero extent
+    if (eltBounds && (eltBounds.width > 0) && (eltBounds.height > 0)) {
+      const board = content.initializeBoard(this.state.elementId, this.handleCreateElement);
+      if (board) {
+        this.handleCreateBoard(board);
+      }
+      const newState = assign({ syncedChanges: content.changes.length },
```

Comment:
Since we're setting state as part of a componentDidUpdate workflow, can we guarantee we only set state if a board has been created? There's part of me that worries that if setting up the board fails for any reason we could end up in an infinite loop... (or am I misremembering my lifecycle?)